File: Swift/maximize-distance-to-closest-person.swift
class Solution {
    func maxDistToClosest(_ seats: [Int]) -> Int {
        var prev = -1
        var res = 1

        for (i, v) in seats.enumerated() {
            if v == 1 {
                if prev < 0 {
                    res = i
                } else {
                    res = max(res, (i - prev) / 2)
                }
                prev = i
            }
        }

        return max(res, seats.count - 1 - prev)
    }
}

File: Python/maximize-distance-to-closest-person.py
class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        prev, res = -1, 1

        for i in range(len(seats)):
            if seats[i]:
                if prev < 0:
                    res = i
                else:
                    res = max(res, (i - prev) // 2)
                prev = i

        return max(res, len(seats) - 1 - prev)
